src/components/lib/kernel/cos_kernel_api.c

```diff
@@ -17,6 +17,51 @@
 #define printd(...)
 #endif
 
+struct pgtbl_lvl_info {
+	unsigned long range;
+	vaddr_t (* round_to_pg)(vaddr_t );
```

Comment:
The purpose here is just : when we are walking the `pgtbl_lvl`, we can easily do the `round_to_pg/round_up_to_pg` operation. If we don't use this, we need add a function which uses a switch structure to do this(I guess this is a little slow operation because of branch prediction). The function pointers here are quite limited that they just return a value here and then do nothing. If we really require that, I can change it here.

Reply:
Fixed.